pkgs/games/0ad/game.nix

```diff
@@ -1,30 +1,115 @@
-{ stdenv, lib, perl, fetchurl, python2
-, pkg-config, spidermonkey_38, boost, icu, libxml2, libpng, libsodium
-, libjpeg, zlib, curl, libogg, libvorbis, enet, miniupnpc
-, openal, libGLU, libGL, xorgproto, libX11, libXcursor, nspr, SDL2
-, gloox, nvidia-texture-tools
-, withEditor ? true, wxGTK ? null
+{stdenv
+,cargo
+,lib
+,perl
+,fetchurl
+,fmt
+,pkg-config
+,boost
+,icu67
+,libxml2
+,libpng
+,libsodium
+,libjpeg
+,zlib
+,curl
+,libogg
+,libvorbis
+,enet
+,miniupnpc
+,openal
+,libGLU
+,libGL
+,xorgproto
+,libX11
+,libXcursor
+,nspr
+,SDL2
+,gloox
+,nvidia-texture-tools
+,libidn
+,rustc
+,rust-cbindgen
+,python3
+,yasm
+,readline
+,autoconf213
+,llvmPackages
+,which
+,zip
+,xorg
+,buildPackages
+,cargo-c
+,withEditor ? true, wxGTK ? null
 }:
-
 assert withEditor -> wxGTK != null;
 
 stdenv.mkDerivation rec {
   pname = "0ad";
-  version = "0.0.23b";
+  version = "0.0.24b";
 
   src = fetchurl {
-    url = "http://releases.wildfiregames.com/0ad-${version}-alpha-unix-build.tar.xz";
-    sha256 = "0draa53xg69i5qhqym85658m45xhwkbiimaldj4sr3703rjgggq1";
+   url = "http://releases.wildfiregames.com/0ad-${version}-alpha-unix-build.tar.xz";
+   sha256 = "1a1py45hkh2cswi09vbf9chikgxdv9xplsmg6sv6xhdznv4j6p1j";
   };
 
-  nativeBuildInputs = [ python2 perl pkg-config ];
+  #this is required so SpiderMonkey builds with (cargo-link is searching impure paths)
+  NIX_ENFORCE_PURITY=0;
+
+  nativeBuildInputs = [
+   autoconf213
+   cargo
+   llvmPackages.llvm
+   llvmPackages.libcxx
+   llvmPackages.libcxxStdenv
+   llvmPackages.libclang
+   python3
+   perl
+   pkg-config
+   rust-cbindgen
+   which
+   yasm
+   zip
+  ];
+
+  patches = [
+    ./rootdir_env.patch
+  ];
+
+  depsBuildBuild = [buildPackages.stdenv.cc];
 
   buildInputs = [
-    spidermonkey_38 boost icu libxml2 libpng libjpeg
-    zlib curl libogg libvorbis enet miniupnpc openal
-    libGLU libGL xorgproto libX11 libXcursor nspr SDL2 gloox
-    nvidia-texture-tools libsodium
-  ] ++ lib.optional withEditor wxGTK;
+   readline
+   cargo-c
+   boost
+   icu67
+   libxml2
+   libpng
+   libjpeg
+   zlib
+   curl
+   libogg
+   libvorbis
+   enet
+   miniupnpc
+   openal
+   libGLU
+   libGL
+   xorgproto
+   libX11
+   libXcursor
+   nspr
+   SDL2
+   gloox
+   nvidia-texture-tools
+   libsodium
+   fmt
+   libidn
+   cargo
+   rustc
+   python3
+   yasm
+   nspr] ++ lib.optional withEditor wxGTK;
```

Comment:
The closing `]` should be after a linebreak.

The following style is commonly found in nixpkgs:
```nix
myListOfThings = [
  somePackage
] + lib.optinal false package;
```